dist/tools/drone-scripts/build_and_test.sh

```diff
@@ -16,9 +16,16 @@ function exec_build_func {
     fi
 
     echo "Begin group ${group}"
+    # spawn new directory for the group
+    mkdir -p "build_tmp/${group}"
+    rsync -a --exclude="build_tmp" . "build_tmp/${group}"
```

Comment:
Hmm, haven't tried hard linking and I do see the advantage. However, I would say that it does make sense to use full copies because:
* the workspaces are then cleanly separated from each other
* copying the files takes only a few seconds (at max) which is a tiny amount of the overall build time for each group